csharp/src/Apache.Arrow.Flight.Sql/PreparedStatement.cs

```diff
@@ -0,0 +1,416 @@
+                {
+                    var writer = new ArrowStreamWriter(memoryStream, parameterBatch.Schema);
+
+                    cancellationToken.ThrowIfCancellationRequested();
+                    await writer.WriteRecordBatchAsync(parameterBatch, cancellationToken).ConfigureAwait(false);
+                    await writer.WriteEndAsync(cancellationToken).ConfigureAwait(false);
+
+                    memoryStream.Position = 0;
+
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    var flightData = new FlightData(
+                        FlightDescriptor.CreateCommandDescriptor(_handle),
+                        ByteString.CopyFrom(memoryStream.ToArray()),
+                        ByteString.Empty,
+                        ByteString.Empty
+                    );
+                    await channel.Writer.WriteAsync(flightData, cancellationToken).ConfigureAwait(false);
+                }
+
+                channel.Writer.Complete();
+            }
+            catch (OperationCanceledException)
+            {
+                channel.Writer.TryComplete(new OperationCanceledException("Task was canceled"));
+            }
+            catch (Exception ex)
+            {
+                channel.Writer.TryComplete(ex);
+            }
+        }, cancellationToken);
+
+        _parameterReader = new FlightServerRecordBatchStreamReader(new ChannelReaderStreamAdapter<FlightData>(channel.Reader));
+        if (task.IsCanceled || cancellationToken.IsCancellationRequested)
+        {
+            return Status.DefaultCancelled;
+        }
+
+        return Status.DefaultSuccess;
+    }
+
+
+
+    /// <summary>
+    /// Executes the prepared statement asynchronously and retrieves the query results as <see cref="FlightInfo"/>.
+    /// </summary>
+    /// <param name="options">The <see cref="FlightCallOptions"/> for the operation, which may include timeouts, headers, and other options for the call.</param>
+    /// <param name="parameterBatch">Optional <see cref="RecordBatch"/> containing parameters to bind before executing the statement.</param>
+    /// <param name="cancellationToken">Optional <see cref="CancellationToken"/> to observe while waiting for the task to complete. The task will be canceled if the token is canceled.</param>
+    /// <returns>A <see cref="Task{FlightInfo}"/> representing the asynchronous operation. The task result contains the <see cref="FlightInfo"/> describing the executed query results.</returns>
+    /// <exception cref="InvalidOperationException">Thrown if the prepared statement is closed or if there is an error during execution.</exception>
+    /// <exception cref="OperationCanceledException">Thrown if the operation is canceled by the <paramref name="cancellationToken"/>.</exception>
+    public async Task<FlightInfo> ExecuteAsync(FlightCallOptions options, RecordBatch parameterBatch, CancellationToken cancellationToken = default)
```

Comment:
Execute should not take a paramterBatch, that should only be for `executeUpdate` https://github.com/apache/arrow/blob/main/cpp/src/arrow/flight/sql/client.cc#L635

Reply:
I've update the implementation